### DISCUSSION TITLE
Overriding the "stream" module

### DISCUSSION THREAD

**Discussion by SourceBoy** [vercel/next.js#47305] [Help]
### Summary

We have a need to use the latest `stream.Readable.fromWeb` and `toWeb` functions to convert the 2 different implementations of streams. Is there a way to override the `stream` module with Node 18 or readable-stream v4?

### Additional information

```js
Use case: Web stream -> Node stream -> Processing -> Web stream
```

**Comment by lstellway** [comment]
I was experiencing the same issue.. 
Casting the type seems to work without issue

```ts
import fs from 'fs';
import { Readable } from 'stream';
import { NextResponse } from 'next/server';

// ...

const filePath = "/path/to/some/file.txt";
const stream = fs.createReadStream(filePath);

// Forcefully cast the type to mitigate type errors
const readable = Readable.toWeb(stream) as ReadableStream<any>;
return new NextResponse(readable);
```
